src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -238,37 +235,24 @@ public MessageImpl createMessage(boolean send, String address, MultiMap headers,
     return msg;
   }
 
-  protected <T> HandlerHolder<T> addRegistration(String address,
-                                                 HandlerRegistration<T> registration,
-                                                 boolean replyHandler,
-                                                 boolean localOnly,
-                                                 Handler<AsyncResult<Void>> completionHandler) {
+  protected <T> HandlerHolder<T> addRegistration(String address, HandlerRegistration<T> registration, boolean replyHandler, boolean localOnly, Handler<AsyncResult<Void>> completionHandler) {
 //    Objects.requireNonNull(registration.getHandler(), "handler");
-    LocalRegistrationResult<T> result = addLocalRegistration(address, registration, replyHandler, localOnly);
-    addRegistration(result.newAddress, result.holder, completionHandler);
-    return result.holder;
+    HandlerHolder<T> holder = addLocalRegistration(address, registration, replyHandler, localOnly);
+    onLocalRegistration(holder, completionHandler);
+    return holder;
   }
 
-  protected <T> void addRegistration(boolean newAddress, HandlerHolder<T> holder, Handler<AsyncResult<Void>> completionHandler) {
+  protected <T> void onLocalRegistration(HandlerHolder<T> handlerHolder, Handler<AsyncResult<Void>> completionHandler) {
```

Comment:
`onLocalUnregistration` has a `Promise<Void>` and this has an `Handler<AsyncResult<Void>>`, it should use also `Promise<Void>` (which actually is simpler to succeed or fail)

Reply:
Done